application-admintools-default/src/main/java/com/xwiki/admintools/script/AdminToolsScriptService.java

```diff
@@ -106,16 +107,18 @@ public String getFilesSection()
     }
 
     /**
-     * Start the execution of the given Admin Tools health check request.
+     * Check if an Admin Tools Health Check job for the wiki from where the request was made exists. If it does, return
+     * the job instance, else create a new Admin Tools health check request for the given wiki and start the execution.
      *
-     * @param healthCheckJobRequest the health check request to be executed.
+     * @param requestId the ID to be used for the request.
      * @return the asynchronous background job that will execute the request.
      */
-    public Job runHealthChecks(HealthCheckJobRequest healthCheckJobRequest)
+    public Job runHealthChecks(List<String> requestId)
```

Comment:
I'm not convinced that the user calling this API should be able to control the entire job ID. Usually jobs of the same type share an ID prefix (e.g. ``export/pdf`` or in your case ``adminTools/healthCheck``). The suffix is what identifies them (among jobs of the same type), and the suffix is most of the time generated automatically. I don't see why the user calling this script service should care about the job ID. They trigger the job and then get the ID from the returned job object.